sfa_dash/blueprints/reports.py

```diff
@@ -132,13 +136,13 @@ class ReportView(BaseView):
     template = 'data/report.html'
 
     def template_args(self):
-        return {'report_metadata': self.metadata}
+        return {'report': report_to_html_body(self.metadata)}
 
     def get(self, uuid):
-        # TODO: use core to buid templates
-        report_request = reports.get_metadata(uuid)
-        if report_request.status_code == 200:
-            self.metadata = report_request.json()
-        else:
-            abort(404)
+        token = oauth_request_session.token['access_token']
+        session = APISession(token)
+        try:
+            report_request = session.get_report(uuid)
+        except HTTPError as e:
+            abort(404);
```

Comment:
@alorenzo175  This isn't complete or anything, but was this sort of inlining what you were suggesting? Or did you mean keeping it in the api_interface module?